src/Database/LSMTree/Internal/MergingRun.hs

```diff
@@ -275,7 +276,7 @@ duplicateRuns (DeRef mr) =
     withMVar (mergeState mr) $ \case
       CompletedMerge r  -> V.singleton <$> dupRef r
       OngoingMerge rs _ -> withActionRegistry $ \reg ->
-        V.mapM (\r -> withRollback reg (dupRef r) releaseRef) rs
+        V.mapMStrict (\r -> withRollback reg (dupRef r) releaseRef) rs
```

Comment:
I think that this one might be superfluous because `dupRef` is already strict enough, but I forgot to take it out